applications/TrilinosApplication/trilinos_space.h

```diff
@@ -167,50 +168,70 @@ class TrilinosSpace
     }
 
     /// rXi = rMij
-
     static void GetColumn(unsigned int j, MatrixType& rM, VectorType& rX)
     {
-        KRATOS_THROW_ERROR(std::logic_error, "GetColumn method is not currently implemented", "")
+        KRATOS_ERROR << "GetColumn method is not currently implemented" << std::endl;
     }
 
-
     ///////////////////////////////// TODO: Take a close look to this method!!!!!!!!!!!!!!!!!!!!!!!!!
     /// rMij = rXi
     //      static void SetColumn(unsigned int j, MatrixType& rM, VectorType& rX){rX = row(rM, j);}
 
     /// rY = rX
-
     static void Copy(MatrixType const& rX, MatrixType& rY)
     {
         rY = rX;
     }
 
     /// rY = rX
-
     static void Copy(VectorType const& rX, VectorType& rY)
     {
         rY = rX;
     }
 
     /// rX * rY
-
     static double Dot(VectorType& rX, VectorType& rY)
     {
         double value;
         double *temp = new double[1];
-        rY.Dot(rX, temp); //it is prepared to handle vectors with multiple components
+        const int sucess = rY.Dot(rX, temp); //it is prepared to handle vectors with multiple components
+        KRATOS_ERROR_IF_NOT(sucess == 0) << "Error computing dot product" <<  std::endl;
         value = temp[0];
         delete [] temp;
         return value;
     }
 
-    /// ||rX||2
+    /// Maximum alue
+    static double Max(const VectorType& rX)
+    {
+        double value;
+        double *temp = new double[1];
```

Comment:
Can you please try with just a double?

This looks very weird

Reply:
Done